patroni/dcs/__init__.py

```diff
@@ -585,50 +796,87 @@ class Cluster(NamedTuple):
 
     @staticmethod
     def empty() -> 'Cluster':
+        """Produce an empty :class:`Cluster` instance."""
         return Cluster(None, None, None, 0, [], None, SyncState.empty(), None, None, None)
 
     def is_empty(self):
-        return self.initialize is None and self.config is None and self.leader is None and self.last_lsn == 0\
-            and self.members == [] and self.failover is None and self.sync.version is None\
-            and self.history is None and self.slots is None and self.failsafe is None and self.workers == {}
+        """Validate definition of all attributes of this :class:`Cluster` instance.
+
+        :returns: ``True`` if the current :class:`Cluster` has unpopulated attributes.
+        """
+        return all((self.initialize is None, self.config is None, self.leader is None, self.last_lsn == 0,
+                    self.members == [], self.failover is None, self.sync.version is None,
+                    self.history is None, self.slots is None, self.failsafe is None, not self.workers))
 
     def __len__(self) -> int:
+        """Implement ``len`` function capability."""
```

Comment:
```suggestion
        """Implement ``len`` function capability.

        .. note::
           This magic method aids in the evaluation of "emptiness" of a ``Cluster`` instance. For example:

           >>> cluster = Cluster.empty()
           >>> len(cluster)
           0
           >>> assert bool(cluster) is False

           >>> cluster = Cluster(None, None, None, 0, [1, 2, 3], None, SyncState.empty(), None, None, None)
           >>> len(cluster)
           1
           >>> assert bool(cluster) is True

           This makes it easier to write ``if cluster`` rather than the longer statement.

        """
```